docs/administration/recent_activity/recent_activity.md

```diff
@@ -49,19 +49,180 @@ This policy is also required by user having a profile to properly view their own
+
+#### Context group
+
+If you log several entries at once, you can group them into a context.
+A context group counts as one item in regard to `activity_logs_limit` configuration and `ActivityLogService::findGroups`'s `$limit` argument.
+
+To open a context group, use `ActivityLogService::prepareContext` which has 2 arguments:
+
+* `$source` is describing (usually through a short identifier) what is triggering the set of actions. For example, some already existing source are `web` (like for actions from the Back Office), `graphql`, `rest` and `migration`.
+* `$description` is an optional more specific contextualisation. For example, `migration` context source is associated with the migration file name in its context description.
+
+To close a context group, use `ActivityLogService::dismissContext`.
+
+In the following example, several actions are logged into one context group, even actions triggered by cascade outside the piece of code:
 
-TODO: Illustrate the use of ActivityLogService(Interface)::build & ActivityLogService(Interface)::save
+- `my_feature`
+    - `init`
+    - `create`
+    - `publish`
+    - `simulate`
+    - `complete`
+
+``` php
+[[= include_file('code_samples/recent_activity/src/Command/ActivityLogContextTestCommand.php', 56, 76) =]]
+```
+
+Context groups can't be nested.
+If a new context is prepared while a context is already grouping log entries, this new context will be ignored.
+To start a new context, make sure to previously dismiss the existing one.
+
+As the context are not really user-friendly, by default, contexts are not shown in the Back Office.
+A context group is folded below its first entry.
+
+#### List
+
+To display your log entry, if your object's PHP class isn't already covered, you'll have to:
+
+* implement `ClassNameMapperInterface` to associate the class name with an identifier,
+* eventually create a `PostActivityListLoadEvent` subscriber if you need to load the object for the template,
+* create a template to display this class log entries.
+
+You can have a template:
+
+* specific to an action on an identifier and placed in `templates/themes/<theme>/activity_log/ui/<identifier>/<action>.html.twig`
+* specific to an identifier and placed in `templates/themes/<theme>/activity_log/ui/<identifier>.html.twig`
+
+Template existence is tested in this order.
+For the same identifier, you could have specific templates for few actions, and a default one for the remaining actions.
+
+Your template can extend `@ibexadesign/activity_log/ui/default.html.twig` and only redefine the `activity_log_description_widget` block for your objects. This default template is itself used if no template is found for the identifier and the action, or the identifier alone. The built-in default template has an empty `activity_log_description_widget` block and display nothing for unknown objects.
```

Comment:
I'd change the information order a bit here. I'd first explain that there is a default template, tell what it displays, and where it is located.

Then I'd explain how it can be overridden.

Reply:
Reordered in 1c47a199f8d8e38f3d0427c9fa179f678c90b718

@bdunogier Is it better?